### ISSUE TITLE
V2: TextField numeric input not working on non-Windows platforms

### ISSUE THREAD

**Issue by cjlotz** [gui-cs/Terminal.Gui#3540]
Hello

For V2, when I'm trying to enter numeric input values for a `TextField` on non-Windows platforms, the values are not accepted by the control.  Is there some specific setting in V2 that I'm missing?  Looking at the changes to the `TextField` between V1 and V2, the V2 `OnProcessKeyDown` contains an additional check around key codes A-Z which is not in the V1 `ProcessKey` method - not sure if this is the cause of the issue and why it works on Windows

```c#
        // Ignore other control characters.
        if (!a.IsKeyCodeAtoZ && (a.KeyCode < KeyCode.Space || a.KeyCode > KeyCode.CharMask))
        {
            return false;
        }
```

**Comment by tznind** [comment]
This sounds like a bug.  But having more information to diagnose would help.

Can you check whether NumLock or Scroll Lock is down when trying to type.  Sometimes that makes a difference (although it really shouldn't).

One thing that would help is to know exactly what the keycode is that is being ignored.

You can trap all key events at application level with a global listener (see below) and then look at `e` in the debugger.  If you could provide a screenshot of what that is showing on the affected OS that would be great.

```csharp
        Application.KeyDown += (s, e) =>
        {
            // Break point here
        };
```

**Comment by cjlotz** [comment]
@tznind Thanks for the reply.  

Following your suggestion, I added a status bar item that logs the key code.  On Windows 11 when I execute this, I see D1, D2... for the digits.  On Linux (Ubuntu 22.04 running in WSL on Windows 11), the same key codes are logged as 17, 18, 19...  

I wasn't sure if running in WSL might be causing the issue, so I compiled this on my Mac as well.  When executing on the Mac I see the same behaviour, i.e. the digits are logged as 17,18,19.  On the Mac I have some additional issues with some other key codes as well.  For example, Backspace is logged as 95 on Mac, but interpreted correctly on Linux and Windows.

Thoughts?

**Comment by cjlotz** [comment]
I've also now verified this with the Keys example on the Mac.  Running the Keys example for the `develop` branch works fine, but running the Keys example from the `v2_develop` branch does not work for digits and some other keys.

**Comment by BDisp** [comment]
@cjlotz can you please test the PR #3542, on `Linux` and `macOS` to see if it working better. On `macOS`, for the `Alt` key to work at all, you need to change the following configuration in `Terminal`. Thanks.

`Terminal > Preferences > Profiles > Keyboard` and just check `Use Option as Meta key`

**Comment by cjlotz** [comment]
@BDisp I can confirm that the fixes work on my Mac and also in Ubuntu 22.04 running in WSL on Windows 11.

**Comment by BDisp** [comment]
This was already fixed on #3542.
